xpk.py

```diff
@@ -2612,6 +2660,38 @@ def get_user_input(input_msg):
   return user_input in ('y', 'yes')
 
 
+def get_node_pools_to_delete(args, existing_node_pool_names, desired_node_pool_names) -> list:
+  """Get list of nodepools to delete from the cluster.
+
+  Args:
+    args: user provided arguments for running the command.
+    existing_node_pool_names: names of nodepools that already exist in the cluster.
+    desired_node_pool_names: names of nodepools that should exist in the cluster.
+
+  Returns:
+    List of nodepool names to delete.
+  """
+  node_pools_to_delete = []
+  desired_node_pool_type = args.tpu_type if args.tpu_type else args.device_type
+  is_requested_device_type_in_cluster = check_device_type_in_cluster(args, desired_node_pool_type)
+  for existing_node_pool_name in existing_node_pool_names:
+    # Deletion logic would leave behind any Pathways CPU nodepools.
+    if existing_node_pool_name.find(f'{args.cluster}-np-') != 0:
+      continue
+
+    # Nodepools will be deleted in two scenarios:
+    # Scenario 1: Cluster exists with 3 nodepools of 'x' device_type and now we are updating
+    # the cluster to 2 nodepools of 'x' device_type. In this case, we will delete
+    # '{args.cluster}-np-2' from the cluster.
+    # Scenario 2: Cluster exists with 2 nodepools of 'x' device_type and now we are updating
+    # the cluster to 2 nodepools of 'y' device_type. In this case, we will delete
+    # '{args.cluster}-np-0' and '{args.cluster}-np-1' from the cluster.
```

Comment:
Can you add in the readme an example of this scenario in the section about recalling cluster create. 

Basically showing: 

How to change your cluster's device_type:

Call xpk clsuter create with a new --device-type=v5-8. The cluster will delete the old node pools.

Reply:
Updated README.